src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -211,11 +211,16 @@ public async Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext cont
             // Open connection and execute query using _queryExecutor
             string queryString = queryBuilder.Build(structure);
 
+            // Global Cache enablement check
             if (runtimeConfig.CanUseCache())
             {
+                // Entity level cache behavior checks
                 bool dbPolicyConfigured = !string.IsNullOrEmpty(structure.DbPolicyPredicatesForOperations[EntityActionOperation.Read]);
+                bool entityCacheEnabled = runtimeConfig.Entities[structure.EntityName].IsCachingEnabled;
```

Comment:
Is caching enabled check applicable only for parent level entities in GraphQL queries - for .e.g. there could be nested entities queried but we are not checking if caching is enabled for those or not. Is this by design?

What happens if 1 is enabled, 1 is not -> does it mean if 1 is not enabled we dont cache at all?

Reply:
Only the top-level entity's `cache` setting is considered which is by-design. 

> What happens if 1 is enabled, 1 is not -> does it mean if 1 is not enabled we dont cache at all?
Request will be cached as long as the entity which has caching enabled, is the top-level entity.